sql/src/test/java/io/crate/operation/scalar/ArrayCatFunctionTest.java

```diff
@@ -0,0 +1,165 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.operation.scalar;
+
+import io.crate.metadata.FunctionIdent;
+import io.crate.metadata.Scalar;
+import io.crate.operation.Input;
+import io.crate.operation.operator.input.ObjectInput;
+import io.crate.planner.symbol.Function;
+import io.crate.planner.symbol.Literal;
+import io.crate.planner.symbol.Symbol;
+import io.crate.testing.TestingHelpers;
+import io.crate.types.ArrayType;
+import io.crate.types.DataType;
+import io.crate.types.DataTypes;
+import org.elasticsearch.ElasticsearchIllegalArgumentException;
+import org.hamcrest.Matchers;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.hamcrest.Matchers.is;
+
+public class ArrayCatFunctionTest extends AbstractScalarFunctionsTest {
+
+    private ArrayCatFunction getFunction(ArrayType... args) {
+        List<DataType> argumentTypes = new ArrayList<>(args.length);
+        for (int i = 0; i < args.length; i++) {
+            argumentTypes.add(args[i]);
+        }
+        ArrayCatFunction function = ((ArrayCatFunction) functions.get(new FunctionIdent(ArrayCatFunction.NAME, argumentTypes)));
+        return function;
+    }
+
+    @Test
+    public void testNormalizeWithValueSymbols() throws Exception {
+        ArrayType type = new ArrayType(DataTypes.INTEGER);
+        ArrayCatFunction function = getFunction(type, type);
+
+        Symbol symbol = function.normalizeSymbol(new Function(function.info(), Arrays.<Symbol>asList(
+                Literal.newLiteral(new Integer[]{10, 20}, type),
+                Literal.newLiteral(new Integer[]{10, 30}, type)
+        )));
+
+        TestingHelpers.assertLiteralSymbol(symbol, new Integer[]{10, 20, 10, 30}, type);
+    }
+
+    @Test
+    public void testNormalizeWithRefs() throws Exception {
+        ArrayType type = new ArrayType(DataTypes.INTEGER);
+        ArrayCatFunction function = getFunction(type, type);
+
+        Function functionSymbol = new Function(function.info(), Arrays.<Symbol>asList(
+                TestingHelpers.createReference("foo", type),
+                Literal.newLiteral(new Integer[]{10, 30}, type)
+        ));
+        Function symbol = (Function) function.normalizeSymbol(functionSymbol);
+        assertThat(symbol, Matchers.sameInstance(functionSymbol));
+    }
+
+    private void assertEval(Object[] expected, Object[] ... args) {
```

Comment:
please don't use objects and guess types, use ``Literal`` as arguments instead of Object[] and provide Array Literals